### DISCUSSION TITLE
pyvista with jupyter lab in a docker does not make plots

### DISCUSSION THREAD

**Discussion by emi0000** [pyvista/pyvista#5261] [Q&A] [answered]
Hi,

I'using pyvista with jupyter lab (version 4.08) in a docker of dolfinx/lab. The operating system is Windows 10
pyvista 0.42.2
vtk 9.3.0
pv.global_theme.jupyter_backend : trame
pv.global_theme.trame.server_proxy_prefix : /proxy/
pv.global_theme.trame.server_proxy_enabled : True

After installing trame, trame-vuetify and trame-vtk all error messages where gone, however no plot appears. I get a frame with the message.
![grafik](https://github.com/pyvista/pyvista/assets/88888989/9ce94a5e-3eab-4127-adce-c8f7624c4af8)

sometimes I get the message that the kernel died. The code I'm using is as follows

      
      import pyvista as pv

      pv.set_jupyter_backend('trame')

      pv.global_theme.trame.server_proxy_enabled=True
      pv.start_xvfb()

      pl = pv.Plotter()
      pl.add_mesh(pv.ParametricKlein())
      pl.show()

Adding off_screen_plot=False and screenshot='file.png' results in the same error. Any ideas?

**Comment by banesullivan** [comment, marked as answer]
Did you install jupyter-server-proxy? ref https://docs.pyvista.org/version/stable/user-guide/jupyter/trame.html#jupyter-server-proxy

**Reply by emi0000** [reply]
thanks, yes I did that already with
pip install jupyter-server-proxy

If I try to set 
      pv.global_theme.trame.server_proxy_enabled=False
I get the message
localhost refused connection (literal translation)

**Reply by emi0000** [reply]
I uninstalled jupyter-server-proxy and reinstalled it, the plot now comes up, In addition I'm getting a red frame with the message

![Figure_0](https://github.com/pyvista/pyvista/assets/88888989/4251e30a-83a1-4006-93be-0372d987c6af)

so seems to work now.

**Reply by banesullivan** [reply]
Sounds like jupyter server proxy wasn't running originally. After installing Jupyter server proxy, it's necessary to relaunch Jupyter lab
